packages/api/src/controllers/stream.ts

```diff
@@ -442,22 +411,42 @@ export async function getRecordingFields(
     return {};
   }
 
+  // Recording V2
+  if (session.version === "v2") {
+    const asset = await db.asset.getBySessionId(session.id);
+    if (!asset) {
+      return { recordingStatus: "waiting" };
+    }
+    const assetWithPlayback = await withPlaybackUrls(config, ingest, asset);
+    const assetPhase = assetWithPlayback.status?.phase;
+    return {
+      recordingStatus:
+        assetPhase == "ready"
+          ? "ready"
+          : assetPhase == "failed"
+          ? "none"
+          : "waiting",
+      recordingUrl: assetWithPlayback.playbackUrl,
+      mp4Url: assetWithPlayback.downloadUrl,
+    };
+  }
+
+  // Backwards-compatibility for Recording V1
   const readyThreshold = Date.now() - USER_SESSION_TIMEOUT;
   const isReady = session.lastSeen > 0 && session.lastSeen < readyThreshold;
   const isUnused = !session.lastSeen && session.createdAt < readyThreshold;
 
   const recordingStatus = isReady ? "ready" : isUnused ? "none" : "waiting";
-  if (!isReady && !forceUrl) {
-    return { recordingStatus };
-  }
-  const recordingUrls = await getRecordingUrls(config, ingest, session);
-  if (!recordingUrls) {
-    return { recordingStatus: "waiting" };
-  }
-  if (!recordingUrls.recordingUrl) {
-    return { recordingStatus: "none" };
-  }
-  return { recordingStatus, ...recordingUrls };
+  const base = pathJoin(
+    ingest,
+    `recordings`,
+    session.lastSessionId ?? session.id
+  );
+  return {
+    recordingStatus: recordingStatus,
+    recordingUrl: pathJoin(base, "index.m3u8"),
+    mp4Url: pathJoin(base, "source.mp4"),
```

Comment:
I think you're only missing a check here to only include the URLs when `isReady || forceUrl`. Otherwise we'd be returning URLs for sessions that don't have a recording for v1 (OTOH this will never be the case anymore as we don't create v1 sessions anymore).

Just in case anyway, you can do it with some ts literal trick as well
```suggestion
    ...(!isReady && !forceUrl ? null : {
      recordingUrl: pathJoin(base, "index.m3u8"),
      mp4Url: pathJoin(base, "source.mp4"),
    })
```

Reply:
ah well spotted thanks